includes/misc-functions.php

```diff
@@ -520,6 +519,47 @@ function _edd_deprecated_function( $function, $version, $replacement = null, $ba
 	}
 }
 
+/**
+ * Marks an argument in a function deprecated and informs when it's been used
+ *
+ * There is a hook edd_deprecated_argument_run that will be called that can be used
+ * to get the backtrace up to what file and function called the deprecated
+ * function.
+ *
+ * The current behavior is to trigger a user error if WP_DEBUG is true.
+ *
+ * This function is to be used in every function that has an argument being deprecated.
+ *
+ * @uses do_action() Calls 'edd_deprecated_argument_run' and passes the argument, function name, what to use instead,
+ *   and the version the function was deprecated in.
+ * @uses apply_filters() Calls 'edd_deprecated_argument_trigger_error' and expects boolean value of true to do
+ *   trigger or false to not trigger error.
+ *
+ * @param string  $argument    The arguemnt that is being deprecated
+ * @param string  $function    The function that was called
+ * @param string  $version     The version of WordPress that deprecated the function
+ * @param string  $replacement Optional. The function that should have been called
+ * @param array   $backtrace   Optional. Contains stack backtrace of deprecated function
+ */
+function _edd_deprected_argument( $argument, $function, $version, $replacement = null, $backtrace = null ) {
+	do_action( 'edd_deprecated_argument_run', $argument, $function, $replacement, $version );
```

Comment:
This actually follows the same usage that the `_edd_deprecated_function` call is using, the order is `$version, $replacement` in the function arguments, then `$replacement, $version` in the filter. I vote we keep it similar to avoid confusion.